src/app/game-state/impossibleTask.service.ts

```diff
@@ -169,12 +169,29 @@ export class ImpossibleTaskService {
     });
   }
 
+  markPriorCompletions(){
+    for (let taskIndex = this.tasks.length - 1; taskIndex >= 0; taskIndex--){
+      if (this.taskProgress[taskIndex].complete){ // Sanity check for the latest complete task
+        this.nextTask = taskIndex + 1;
+        while(taskIndex >= 0){
```

Comment:
I don't think you need this extra loop, since you'll be hitting the rest of them in the outer loop. Just set a boolean flag that you've found one and then set complete =true if that found flag is set.

Reply:
That ignores the setting of this.nextTask above that loop, which is the primary issue, and I'd have to do a secondary loop to check values for that if I do as suggested. The inner loop allows me to set the nextTask at that point without having to do a bunch of other stuff and finishes out the outer loop.

Reply:
Wait no, you're right, I'm reading this wrong.